File: Array&String/Python/thousand_separator.py
class Solution:
    def thousandSeparator(self, n: int) -> str:
        n = str(n)
        le = len(n)
        rem = le % 3
        if le < 4: return n
        ans = n[:rem]
        r = '.'.join(n[i:i+3] for i in range(rem, le, 3))
        return f"{ans}." + r if ans else r

    # shorter
    def thousandSeparator(self, n: int) -> str:
        return '{:,}'.format(n).replace(',', '.') if n >= 0 else '-{:,}'.format(-n).replace(',', '.')

    # shorter using indexing
    def thousandSeparator(self, n: int) -> str:
        s, res = str(n), ""
        n = len(s)
        for i in range(n):
            if i > 0 and (n - i) % 3 == 0: res += "."
            res += s[i]
        return res

File: Array&String/CPP/thousand_separator.cpp
#include <format>
#include <iostream>
#include <string>
using namespace std;

class Solution {
    public:
        string thousandSeparator(int n) {
            if (n < 1000) return to_string(n);
            string nstr = to_string(n);
            int le = nstr.size();
            int rem = le % 3;
            string sliced = nstr.substr(rem), ans = "";
            int sliced_len = sliced.size();
            for (int i = 0; i < sliced_len; i += 3) {
                ans += format(".{}", sliced.substr(i, 3));
            }
            return rem > 0? nstr.substr(0, rem) + ans : ans.substr(1);
        }

        // shorter
        string thousandSeparator(int n) {
            string s = to_string(n), res;
            int len = s.size();
            for (int i = 0; i < len; ++i) {
                if (i > 0 && (len - i) % 3 == 0) res += '.';
                res += s[i];
            }
            return res;
        }
    };